doc/GENERATOR.md

```diff
@@ -7,17 +7,165 @@ Simply put, a generator is:
+- a `flag` property can be provided if you want to allow your parameter to be deactivated. In the `GeneratorMeta` registration process, the associated boolean `Q_PROPERTY` name is understood as **the value of `propName` prefixed by `flag`, camel-cased**. For example, the flag's property name for `"propName": "myProperty"` would be `flagMyProperty`;
+- a `flagDefault` boolean property (only necessary if `flag` is set to `true`, to determine) that defines the default value of the flag checkbox.
+
+A static hash map containing the default values for field declarations is defined in `GeneratorMeta::fieldDefaults`.
+
+### `help` property
+This property is defined as an array of objects, each representing its own `SubRack` instance inserted at the bottom of the "Generator info" `Rack`. Again, for each object, the `title` and `html_path` properties are required. Make sure the path is relative to the `meta.json` file's parent directory.
+
+The contents of the HTML file follow [QML's StyledText text format standard](https://doc.qt.io/qt-5/qml-qtquick-text.html#textFormat-prop).
+
+### `enumLabels` property
+This optional object is used as a string map for `enum`s that might be used both in the C++ `Generator` sub-class and as exposed Qt properties.
+
+For each `enum`, add a key-value pair where:
+- the key corresponds to the Pascal-case `enum` name;
+- the value corresponds to a string list, of which each element correlates to its respectively ordered `enum` item.
+
+For further information on how to prepare `enum`s for full-stack use, refer to "Additional configuration for `enum`s" below.
+
+## Field type options
+Here are the possible values for `type`, followed by the required properties you need to provide for these types:
+- `"number"`: int value field; `"min": int, "max": int`;
+- `"slider"`: float value field; `"min": float, "max": float`;
+- `"select"`: choice selection field; `"enumName": int`;
+- `"text"`: single-line text field; `"placeholder": string`;
+- `"area"`: multi-line text field; `"placeholder": string`.<
+
+Please note that `select` fields accept an **integer value**, and not a string value; less so an actual enum identifier. This is to facilitate parsing in C++, which is very strict on the ways you can access enum values. When defining a default value for such a field, make sure that the index you input here corresponds to the index of the enum you're trying to target back in C++.
+
+## Additional .qrc procedures
+Currently, the only way to include generators is by burning them into the application when it is built, rather than have them stand as all-in-1 external packages. As such, any new file that is added inside of `generators` should imperatively be included in the Qt resource system file (`qml.qrc`). Errors *will* otherwise occur at runtime.
```

Comment:
Rephrase "Currently, the only way to include generators is by burning them into the application when it is built, rather than have them stand as all-in-1 external packages. As such, any new file that is added inside of `generators` should imperatively be included in the Qt resource system file (`qml.qrc`). Errors *will* otherwise occur at runtime." to "All the generators are bundled with the application at build time. Any metadata file for the generators must be added to the Qt resource file (`qml.qrc`)."